### ISSUE TITLE
Default install publicly accessible

### ISSUE THREAD

**Issue by hyperknot** [FusionAuth/fusionauth-issues#2040]
### Description
Once you run the ZIP install on macOS 
```
sh -c "curl -fsSL https://raw.githubusercontent.com/FusionAuth/fusionauth-install/master/install.sh | sh"
```

And run ./startup.sh, the resulting service is publicly accessible.

I can access it on 0.0.0.0:9011 and I can also access it on my_ip_address:9011. I can access it from an external computer.

Since many developers are working from co-working places, cafes, etc. this is quite a bad default setting.

Or, in case the default is supposed to be a non-public IP, please take this as a bug report.

I found this line in the default `fusionauth.properties`, which makes me think that it should be 127.0.0.1.

```
# Optionally specify an addressable URL or IP address to access this node from other FusionAuth nodes.
# - In most cases this should be left empty and it will be determined at runtime.
# - Ideally this is a site local IP address that is not publicly routable but accessible from other FusionAuth nodes.
fusionauth-app.url=
```

Also, in this very config setting, I tried using `127.0.0.1` as well as `localhost`, but both resulted in an Java exception when starting up.

In summary:
- defaults are bad
- or defaults are good but there is a bug somewhere
- it is not possible to manually fix it by setting fusionauth-app.url

**Comment by robotdan** [comment]
Internal:

We could add a configuration to pass into the HTTP server to restrict the interfaces we bind on startup. Currently we bind to `0.0.0.0` which binds to all available addresses. Then in FusionAuth we could default this to `127.0.01` or perhaps all local and site-local addresses.

**Comment by robotdan** [comment]
To note, `fusionauth-app.url` is used by other FusionAuth nodes in a clustered configuration and does not change how the server binds on startup.
